modules/externalipam/ipam_validator.rb

```diff
@@ -0,0 +1,49 @@
+require 'proxy/validations'
+require 'externalipam/externalipam'
+require 'resolv'
+
+# Module containing validation methods for use by all External IPAM provider implementations
+module Proxy::Ipam::IpamValidator
+  include ::Proxy::Validations
+  include Proxy::Ipam::IpamHelper
+
+  def validate_required_params!(required_params, params)
+    err = []
+    required_params.each do |param|
+      unless params[param.to_sym]
+        err.push errors[param.to_sym]
+      end
+    end
+    raise Proxy::Validations::Error, err unless err.empty?
+  end
+
+  def validate_ip!(ip)
+    good_ip = ip =~ Regexp.union([Resolv::IPv4::Regex, Resolv::IPv6::Regex])
+    raise Proxy::Validations::Error, ERRORS[:bad_ip] if good_ip.nil?
+    ip
+  end
+
+  def validate_cidr!(address, prefix)
+    cidr = "#{address}/#{prefix}"
+    network = IPAddr.new(cidr).to_s
+    if IPAddr.new(cidr).to_s != IPAddr.new(address).to_s
+      raise Proxy::Validations::Error, "Network address #{address} should be #{network} with prefix #{prefix}"
+    end
+    cidr
+  rescue IPAddr::Error => e
+    raise Proxy::Validations::Error, e.to_s
+  end
+
+  def validate_ip_in_cidr!(ip, cidr)
+    unless IPAddr.new(cidr).include?(IPAddr.new(ip))
+      raise Proxy::Validations::Error.new, "IP #{ip} is not in #{cidr}"
+    end
+  end
+
+  def validate_mac!(mac)
+    unless /^([0-9a-fA-F]{2}[:]){5}[0-9a-fA-F]{2}$/i =~ mac
+      raise Proxy::Validations::Error.new, ERRORS[:bad_mac]
+    end
+    mac
+  end
+end
```

Comment:
We already have `validate_mac` and ip in the codebase, would you mind extracting the current usage into some helper and then reuse that helper here too? No CIDR validations yet.

Reply:
Have not done this yet, but will in the next push